durabletask-azuremanaged/durabletask/azuremanaged/internal/access_token_manager.py

```diff
@@ -4,47 +4,43 @@
 from datetime import datetime, timedelta, timezone
 from typing import Optional
 import durabletask.internal.shared as shared
+from azure.core.credentials import TokenCredential
 
 # By default, when there's 10minutes left before the token expires, refresh the token
 class AccessTokenManager:
-    def __init__(self, refresh_interval_seconds: int = 600, use_managed_identity: bool = False, client_id: str = None):
-        self.scope = "https://durabletask.io/.default"
-        self.refresh_interval_seconds = refresh_interval_seconds
-        self._use_managed_identity = use_managed_identity
-        self._client_id = client_id
+    def __init__(self, refresh_interval_seconds: int = 600, token_credential: TokenCredential = None):
+        self._scope = "https://durabletask.io/.default"
+        self._refresh_interval_seconds = refresh_interval_seconds
         self._logger = shared.get_logger("token_manager")
 
-        # Choose the appropriate credential based on use_managed_identity
-        if self._use_managed_identity:
-            if not self._client_id:
-                self._logger.debug("Using System Assigned Managed Identity for authentication.")
-                self.credential = ManagedIdentityCredential()
-            else:
-                self._logger.debug("Using User Assigned Managed Identity for authentication.")
-                self.credential = ManagedIdentityCredential(client_id=self._client_id)
+        # Choose the appropriate credential. 
+        # Both TokenCredential and DefaultAzureCredential get_token methods return an AccessToken
+        if token_credential:
+            self._logger.debug("Using user provided token credentials.")
+            self._credential = token_credential
         else:
-            self.credential = DefaultAzureCredential()
+            self._credential = DefaultAzureCredential()

```

Comment:
Let's not specify any default credential. The user *must* provide their own credential. If they explicitly provide `None` as their credential, then we should skip all the token manager stuff entirely and make unauthenticated requests (which is valid in local testing setups).